src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/NewAFCalculator.java

```diff
@@ -0,0 +1,165 @@
+package org.broadinstitute.hellbender.tools.walkers.genotyper.afcalc;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.GenotypesContext;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.commons.math3.util.MathArrays;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.GenotypeAlleleCounts;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.GenotypeLikelihoodCalculator;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.GenotypeLikelihoodCalculators;
+import org.broadinstitute.hellbender.utils.Dirichlet;
+import org.broadinstitute.hellbender.utils.IndexRange;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
+ */
+public final class NewAFCalculator extends AFCalculator {
+    private static final GenotypeLikelihoodCalculators GL_CALCS = new GenotypeLikelihoodCalculators();
+    private static final double THRESHOLD_FOR_ALLELE_COUNT_CONVERGENCE = 0.1;
+    private static final int HOM_REF_GENOTYPE_INDEX = 0;
+
+    private final double refPseudocount;
+    private final double snpPseudocount;
+    private final double indelPseudocount;
+    private final int defaultPloidy;
+
+
+    public NewAFCalculator(final double refPseudocount, final double snpPseudocount, final double indelPseudocount, final int defaultPloidy) {
+        this.refPseudocount = refPseudocount;
+        this.snpPseudocount = snpPseudocount;
+        this.indelPseudocount = indelPseudocount;
+        this.defaultPloidy = defaultPloidy;
+    }
+
+    public AFCalculationResult getLog10PNonRef(final VariantContext vc) {
+        return getLog10PNonRef(vc, defaultPloidy, 0, null);
+    }
+    //TODO: this should be a class of static methods once the old AFCalculator is gone.
+    /**
+     * Compute the probability of the alleles segregating given the genotype likelihoods of the samples in vc
+     *
+     * @param vc the VariantContext holding the alleles and sample information.  The VariantContext
+     *           must have at least 1 alternative allele
+     * @param refSnpIndelPseudocounts a total hack.  A length-3 vector containing Dirichlet prior pseudocounts to
+     *                                be given to ref, alt SNP, and alt indel alleles.  Hack won't be necessary when we destroy the old AF calculators
+     * @return result (for programming convenience)
+     */
+    @Override
+    public AFCalculationResult getLog10PNonRef(final VariantContext vc, final int defaultPloidy, final int maximumAlternativeAlleles, final double[] refSnpIndelPseudocounts) {
```

Comment:
Yeah, I apologize for the hacks required to make it conform to `AFCalculator`.  Hopefully we'll be able to eliminate `StateTracker` and friends soon and clean this all up.